### PR TITLE
build: Update shared api-extractor lint config to enable enforcement

### PR DESCRIPTION
**Pull request by Josmithr**
Also removes package-local overrides, which are no longer necessary

### CODE REVIEW

**Comment by Josmithr** [comment]
Waiting for the next release to be cut before merging.

**Review by alexvy86 — APPROVED** [review]

**Comment by tylerbutler** [comment]
nit on the pr title: maybe api-extractor instead of lint? Not sure that's better or more accurate.

File: experimental/dds/tree2/api-extractor-lint.json
Comment (Josmithr) [inline experimental/dds/tree2/api-extractor-lint.json:5]: This was necessary - tree2 was the only package explicitly not linting cross-package compat.

**Pull request merged**
